src/lib.rs

```diff
@@ -153,6 +153,26 @@ pub enum MemberOrigin {
     Offset(u32),
 }
 
+/// The interpolation qualifier of a binding or struct field.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(Serialize))]
+#[cfg_attr(feature = "deserialize", derive(Deserialize))]
+pub enum Interpolation {
+    /// Indicates that linear, non-perspective, correct
+    //// interpolation must be used.
+    NoPerspective,
```

Comment:
perhaps we can call it `Linear` instead?

Reply:
It's called `NoPerspective` in most of the literature (spirv, glsl, etc). Happy to rename it though,.

Reply:
I believe it's only called that because the human-written languages have a default, which is perspective. Once you don't have a default, the name `Linear` starts to make much more sense. And in IR we don't need to have defaults.